logstash-core/src/main/java/org/logstash/instrument/witness/EventsWitness.java

```diff
@@ -0,0 +1,205 @@
+     *
+     * @param count the count to increment by
+     */
+    public void filtered(long count) {
+        filtered.increment(count);
+        dirty = true;
+    }
+
+    /**
+     * Get a reference to associated forgetter.
+     *
+     * @return the associate {@link Forgetter}
+     */
+    public Forgetter forget() {
+        return forgetter;
+    }
+
+
+    /**
+     * increment the in count by 1
+     */
+    public void in() {
+        in.increment();
+        dirty = true;
+    }
+
+    /**
+     * increment the in count
+     *
+     * @param count the number to increment by
+     */
+    public void in(long count) {
+        in.increment(count);
+        dirty = true;
+    }
+
+    /**
+     * increment the out count by 1
+     */
+    public void out() {
+        out.increment();
+        dirty = true;
+    }
+
+    /**
+     * increment the count
+     *
+     * @param count the number by which to increment by
+     */
+    public void out(long count) {
+        out.increment(count);
+        dirty = true;
+    }
+
+    /**
+     * Get a reference to associated snitch to get discrete metric values.
+     *
+     * @return the associate {@link Snitch}
+     */
+    public Snitch snitch() {
+        return snitch;
+    }
+
+    /**
+     * Add to the existing queue push duration
+     *
+     * @param durationToAdd the duration to add
+     */
+    public void queuePushDuration(long durationToAdd) {
+        queuePushDuration.increment(durationToAdd);
+        dirty = true;
+    }
+
+    /**
+     * The snitch for the {@link EventsWitness}. Allows to read discrete metrics values.
+     */
+    public static class Snitch {
+
+        private final EventsWitness witness;
+
+        Snitch(EventsWitness witness) {
+            this.witness = witness;
+        }
+
+        /**
+         * Gets the duration of the events.
+         * @return the events duration.
+         */
+        public long duration() {
+            return witness.duration.getValue();
+        }
+
+        /**
+         * Gets the filtered events count.
+         *
+         * @return the count of the filtered events.
+         */
+        public long filtered() {
+            return witness.filtered.getValue();
+        }
+
+        /**
+         * Gets the in events count.
+         *
+         * @return the count of the events in.
+         */
+        public long in() {
+            return witness.in.getValue();
+        }
+
+        /**
+         * Gets the out events count.
+         *
+         * @return the count of the events out.
+         */
+        public long out() {
+            return witness.out.getValue();
+        }
+
+        /**
+         * Gets the duration of the queue push
+         * @return the queue push duration.
+         */
+        public long queuePushDuration() {
+            return witness.queuePushDuration.getValue();
+        }
+
+    }
+
+    /**
+     * The forgetter for events.
+     */
+    public static class Forgetter {
```

Comment:
I agree that this comes down to style in large part.

One potentially maintenance issue I can see here is that `thing.forget()` is a verb, but it actually just returns the `Forgetter`, I can foresee that being the cause of bugs if the `.all` is left out.

I realize that this is part of the fluid DSL style here, but I think I'd prefer my nouns to be nouns and my verbs to be verbs. I'm fine with or w/o the Forgetters, but I do think that `thing.forgetter.forgetAll()` and `thing.forgetAll()` are more ergonomic in terms of preventing bugs.